precompiles/staking/tx_test.go

```diff
@@ -14,6 +15,299 @@ import (
+		gas         uint64
+		postCheck   func(data []byte)
+		expError    bool
+		errContains string
+	}{
+		{
+			"fail - empty input args",
+			func() []interface{} {
+				return []interface{}{}
+			},
+			200000,
+			func(data []byte) {},
+			true,
+			fmt.Sprintf(cmn.ErrInvalidNumberOfArgs, 7, 0),
+		},
+		{
+			"fail - different origin than delegator",
+			func() []interface{} {
+				differentAddr := evmosutiltx.GenerateAddress()
+				return []interface{}{
+					description,
+					commission,
+					minSelfDelegation,
+					differentAddr,
+					sdk.ValAddress(differentAddr.Bytes()).String(),
+					pubkey,
+					value,
+				}
+			},
+			200000,
+			func(data []byte) {},
+			true,
+			"is not the same as delegator address",
+		},
+		{
+			"fail - invalid description",
+			func() []interface{} {
+				return []interface{}{
+					"",
+					commission,
+					minSelfDelegation,
+					delegatorAddress,
+					validatorAddress,
+					pubkey,
+					value,
+				}
+			},
+			200000,
+			func(data []byte) {},
+			true,
+			"invalid description",
+		},
+		{
+			"fail - invalid commission",
+			func() []interface{} {
+				return []interface{}{
+					description,
+					"",
+					minSelfDelegation,
+					delegatorAddress,
+					validatorAddress,
+					pubkey,
+					value,
+				}
+			},
+			200000,
+			func(data []byte) {},
+			true,
+			"invalid commission",
+		},
+		{
+			"fail - invalid min self delegation",
+			func() []interface{} {
+				return []interface{}{
+					description,
+					commission,
+					"",
+					delegatorAddress,
+					validatorAddress,
+					pubkey,
+					value,
+				}
+			},
+			200000,
+			func(data []byte) {},
+			true,
+			"invalid amount",
+		},
+		{
+			"fail - invalid delegator address",
+			func() []interface{} {
+				return []interface{}{
+					description,
+					commission,
+					minSelfDelegation,
+					"",
+					validatorAddress,
+					pubkey,
+					value,
+				}
+			},
+			200000,
+			func(data []byte) {},
+			true,
+			"invalid delegator address",
+		},
+		{
+			"fail - invalid validator address",
+			func() []interface{} {
+				return []interface{}{
+					description,
+					commission,
+					minSelfDelegation,
+					delegatorAddress,
+					1205,
+					pubkey,
+					value,
+				}
+			},
+			200000,
+			func(data []byte) {},
+			true,
+			"invalid type for",
+		},
+		{
+			"fail - invalid pubkey",
+			func() []interface{} {
+				return []interface{}{
+					description,
+					commission,
+					minSelfDelegation,
+					delegatorAddress,
+					validatorAddress,
+					1205,
+					value,
+				}
+			},
+			200000,
+			func(data []byte) {},
+			true,
+			"invalid type for",
+		},
+		{
+			"fail - invalid pubkey",
+			func() []interface{} {
+				return []interface{}{
+					description,
+					commission,
+					minSelfDelegation,
+					delegatorAddress,
+					validatorAddress,
+					"bHVrZQ=", // base64.StdEncoding.DecodeString error
+					value,
+				}
+			},
+			200000,
+			func(data []byte) {},
+			true,
+			"",
+		},
```

Comment:
Would be good to assert the specific error message here instead of checking for the empty string 🙏 

```suggestion
		{
			"fail - pubkey decoding error",
			func() []interface{} {
				return []interface{}{
					description,
					commission,
					minSelfDelegation,
					delegatorAddress,
					validatorAddress,
					"bHVrZQ=", // base64.StdEncoding.DecodeString error
					value,
				}
			},
			200000,
			func(data []byte) {},
			true,
			"", // <---- add meaningful error string here to check
		},
```

Reply:
Good idea! done.